src/core/entity.js

```diff
@@ -57,19 +52,44 @@ class Entity {
       37: 'r_l'
     };
 
-    math.vec3.subtract(this.forward, look_at, this.position);
-    math.vec3.cross(this.right, this.forward, up);
-    math.vec3.cross(this.up, this.right, this.forward);
-
-    math.vec3.normalize(this.forward, this.forward);
-    math.vec3.normalize(this.right, this.right);
-    math.vec3.normalize(this.up, this.up);
-
     if (this.vertices && this.indices && this.normals) {
       this.create_buffers();
     }
   }
 
+  look_at(vec, up = this.up) {
+    math.mat4.target_to(this.local_matrix, this.position, vec, up);
+  }
+
+  get left() {
```

Comment:
The current coordinate system (Y = up) is left-handed, so it makes sense to rename `right` to `left`. Note that this isn't used anywhere else in the code right now.

Reply:
Doooooo we need to keep this vector then? Since you've get rid of `Entity::realign` and all the computations are handled without it, I believe it became useless. Do you?